fpm/src/fpm_strings.f90

```diff
@@ -19,6 +22,10 @@ module fpm_strings
     procedure :: fnv_1a_string_t
 end interface fnv_1a
 
+interface str
+    module procedure str_int, str_real, str_real_n, str_logical
+end interface
```

Comment:
`str` is likely to be used at some point as a name for a temporary string and it already is:

```f90
src/fpm/toml.f90
87:        character(len=:), allocatable :: str
94:                call get_value(children, ilist, str, stat=stat)
99:                call move_alloc(str, list(ilist)%s)
103:            call get_value(table, key, str, stat=stat)
108:            if (allocated(str)) then
110:                call move_alloc(str, list(1)%s)

src/fpm_strings.f90
97:elemental pure function lower(str,begin,end) result (string)
102:    character(*), intent(In)     :: str
103:    character(len(str))          :: string
107:    string = str
114:    iend = len_trim(str)
120:        select case (str(i:i))
122:            string(i:i) = char(iachar(str(i:i))+32)     ! change letter to miniscule
```

If it should be short, how about `char` instead?

Reply:
There doesn't seem to be a conflict though. If there is, I'd rather rename temporary vars. IMO `str` is most appropriate for this function.

Reply:
Ultimately this will go into `stdlib`, but I just put it here for now, since I needed it.